flink-ml-lib/src/main/java/org/apache/flink/ml/classification/naivebayes/NaiveBayesModel.java

```diff
@@ -0,0 +1,213 @@
+                            this.getClass().getSimpleName(),
+                            outputTypeInfo,
+                            new PredictLabelOperator(featuresCol, broadcastModelKey));
+                };
+        DataStream<Row> output =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(input), broadcastMap, function);
+
+        Table outputTable = tEnv.fromDataStream(output);
+
+        return new Table[] {outputTable};
+    }
+
+    @Override
+    public void save(String path) throws IOException {
+        String dataPath = ReadWriteUtils.getDataPath(path);
+        FileSink<NaiveBayesModelData> sink =
+                FileSink.forRowFormat(
+                                new Path(dataPath), new NaiveBayesModelData.ModelDataEncoder())
+                        .withRollingPolicy(OnCheckpointRollingPolicy.build())
+                        .withBucketAssigner(new BasePathBucketAssigner<>())
+                        .build();
+        NaiveBayesModelData.getModelDataStream(modelDataTable).sinkTo(sink);
+
+        ReadWriteUtils.saveMetadata(this, path);
+    }
+
+    public static NaiveBayesModel load(StreamExecutionEnvironment env, String path)
+            throws IOException {
+        Source<NaiveBayesModelData, ?, ?> source =
+                FileSource.forRecordStreamFormat(
+                                new NaiveBayesModelData.ModelDataStreamFormat(),
+                                ReadWriteUtils.getDataPaths(path))
+                        .build();
+        NaiveBayesModel model = ReadWriteUtils.loadStageParam(path);
+        DataStream<NaiveBayesModelData> modelData =
+                env.fromSource(source, WatermarkStrategy.noWatermarks(), "modelData");
+        model.setModelData(NaiveBayesModelData.getModelDataTable(modelData));
+
+        return model;
+    }
+
+    @Override
+    public Map<Param<?>, Object> getParamMap() {
+        return paramMap;
+    }
+
+    @Override
+    public NaiveBayesModel setModelData(Table... inputs) {
+        Preconditions.checkArgument(inputs.length == 1);
+        modelDataTable = inputs[0];
+        return this;
+    }
+
+    @Override
+    public Table[] getModelData() {
+        return new Table[] {modelDataTable};
+    }
+
+    private static class PredictLabelOperator
+            extends AbstractUdfStreamOperator<Row, AbstractRichFunction>
+            implements OneInputStreamOperator<Row, Row> {
+        private final String featuresCol;
+        private final String broadcastModelKey;
+
+        public PredictLabelOperator(String featuresCol, String broadcastModelKey) {
+            super(new AbstractRichFunction() {});
+            this.featuresCol = featuresCol;
+            this.broadcastModelKey = broadcastModelKey;
+        }
+
+        @Override
+        public void processElement(StreamRecord<Row> streamRecord) {
+            Row inputRow = streamRecord.getValue();
+            NaiveBayesModelData modelData =
```

Comment:
nits: can you cache the `modelData` instead of accessing repeatly from the broadcast context?

Reply:
OK. I'll do it.